client/client.go

```diff
@@ -574,6 +574,56 @@ func (r *NotaryRepository) GetDelegationRoles() ([]*data.Role, error) {
 	return allDelegations, nil
 }
 
+// RoleWithSignatures is a Role with its associated signatures
+type RoleWithSignatures struct {
+	Signatures []data.Signature
+	data.Role
+}
+
+// GetRepoRoleMetaInfo returns a list of RoleWithSignatures objects for this repo
+// This represents the latest metadata for each role in this repo
+func (r *NotaryRepository) GetRepoRoleMetaInfo() ([]RoleWithSignatures, error) {
```

Comment:
I might shorten the function name to simply `ListRoles` (for consistency with `ListTargets` and maybe it can take a single argument `withDetail bool`. If `withDetail` is false, we only populate the basic role info (i.e. the content of a `data.Role` object translated into a `RoleWithSignatures` object). If `withDetail` is `true`, we return the public keys and signatures data too (which I believe is the current behaviour).

I don't think there's a need for a single function that that returns combined targets and roles info. Yes we could make it marginally easier for others to get one big report with a single function, but I don't think it's worthwhile over having 2 functions with clearly defined responsibilities.